crates/eth-rpc/src/rpc.rs

```diff
@@ -1,34 +1,49 @@
+use std::collections::HashMap;
+use std::marker::PhantomData;
 use std::sync::Arc;
 
-use jsonrpsee::RpcModule;
+use jsonrpsee::core::Error;
+use jsonrpsee::{Methods, RpcModule};
 use kakarot_rpc_core::client::api::KakarotEthApi;
 use starknet::providers::Provider;
 
 use crate::api::alchemy_api::AlchemyApiServer;
 use crate::api::eth_api::EthApiServer;
-use crate::servers::eth_rpc;
+use crate::servers::alchemy_rpc::AlchemyRpc;
 use crate::servers::eth_rpc::KakarotEthRpc;
-use crate::servers::token_rpc::{self, TokenRpc};
 
-pub struct KakarotRpc<P: Provider + Send + Sync + 'static> {
-    eth_rpc_module: RpcModule<KakarotEthRpc<P>>,
-    token_rpc_module: RpcModule<TokenRpc<P>>,
+/// Represents RPC modules that are supported by reth
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash)]
+pub enum KakarotRpcModule {
+    Eth,
+    Alchemy,
 }
 
-impl<P: Provider + Send + Sync + 'static> KakarotRpc<P> {
+pub struct KakarotRpcModuleBuilder<P: Provider + Send + Sync + 'static> {
+    modules: HashMap<KakarotRpcModule, Methods>,
+    _phantom: PhantomData<P>,
+}
+
+impl<P: Provider + Send + Sync + 'static> KakarotRpcModuleBuilder<P> {
     pub fn new(kakarot_client: Arc<Box<dyn KakarotEthApi<P>>>) -> Self {
-        let eth_rpc_module = eth_rpc::KakarotEthRpc::new(kakarot_client.clone()).into_rpc();
-        let token_rpc_module = token_rpc::TokenRpc::new(kakarot_client).into_rpc();
+        let eth_rpc_module = KakarotEthRpc::new(kakarot_client.clone()).into_rpc();
+        let alchemy_rpc_module = AlchemyRpc::new(kakarot_client).into_rpc();
+
+        let mut modules: HashMap<KakarotRpcModule, Methods> = HashMap::new();
+
+        modules.insert(KakarotRpcModule::Eth, eth_rpc_module.into());
+        modules.insert(KakarotRpcModule::Alchemy, alchemy_rpc_module.into());
 
-        Self { eth_rpc_module, token_rpc_module }
+        Self { modules, _phantom: PhantomData }
     }
 
-    pub fn rpc_module(&self) -> RpcModule<()> {
+    pub fn rpc_module(&self) -> Result<RpcModule<()>, Error> {
```

Comment:
i guess following the builder pattern, this fn should be name `build` or `build_rpc_module`

Reply:
makes sense, refactored to `build_rpc_module`